cyclonedx/model/component.py

```diff
@@ -759,23 +809,24 @@ def __init__(self, *, name: str, component_type: ComponentType = ComponentType.L
                 'standard', DeprecationWarning
             )
             if not licenses:
-                self.licenses = [LicenseChoice(license_expression=license_str)]  # type: ignore
+                self.licenses = [LicenseChoice(expression=license_str)]  # type: ignore
 
         self.__dependencies: "SortedSet[BomRef]" = SortedSet()
 
-    @property
-    def type(self) -> ComponentType:
+    @property  # type: ignore[misc]
+    @serializable.xml_attribute()
+    def type_(self) -> ComponentType:
         """
         Get the type of this Component.
 
         Returns:
             Declared type of this Component as `ComponentType`.
         """
-        return self._type
+        return self._type_
 
-    @type.setter
-    def type(self, component_type: ComponentType) -> None:
-        self._type = component_type
+    @type_.setter
+    def type_(self, component_type: ComponentType) -> None:
```

Comment:
:memo: breaking change.


but why???

Reply:
Avoid pythonic keyword `type`

Reply:
i would say this does not apply on a data model class.
let's overrule this, if it was a rule.